### ISSUE TITLE
Beta modules plan fails for module.gke.data.google_compute_zones.available, 400 invalid_grant

### ISSUE THREAD

**Issue by jpresky-seegrid** [terraform-google-modules/terraform-google-kubernetes-engine#1083]
### TL;DR

I get the following when I do a plan for this module:
terraform-google-modules/kubernetes-engine/google//modules/beta-private-cluster-update-variant

│ Error: Get "https://compute.googleapis.com/compute/beta/projects/project_name/zones?alt=json&filter=&prettyPrint=false": oauth2: cannot fetch token: 400 Bad Request
│ Response: {
│   "error": "invalid_grant",
│   "error_description": "Bad Request"
│ }
│
│   with module.gke.data.google_compute_zones.available,
│   on .terraform/modules/gke/modules/beta-private-cluster-update-variant/main.tf line 22, in data "google_compute_zones" "available":
│   22: data "google_compute_zones" "available" {

I have tried other beta modules too and get the same result. I have tried planning with the non beta modules such as terraform-google-modules/kubernetes-engine/google//modules/private-cluster-update-variant and that works. I also dug around and saw that the non beta module makes a reference to module.gke.data.google_compute_zones.available and that works with no problems. Not sure why the beta modules behave this way.

### Expected behavior

I expect the beta modules to be able to plan

### Observed behavior

I get the error above. The non-beta modules work even though they data block the same resource that is causing the error. 

### Terraform Configuration

```hcl
Terraform v1.0.11
on darwin_amd64
+ provider registry.terraform.io/hashicorp/external v2.1.0
+ provider registry.terraform.io/hashicorp/google v3.90.0
+ provider registry.terraform.io/hashicorp/google-beta v3.90.0
+ provider registry.terraform.io/hashicorp/kubernetes v2.6.1
+ provider registry.terraform.io/hashicorp/null v3.1.0
+ provider registry.terraform.io/hashicorp/random v3.1.0
➜  terraform-gke git:(jpresky/CO-257-gke-tf) ✗ terraform providers

Providers required by configuration:
.
├── provider[registry.terraform.io/hashicorp/google]
├── provider[registry.terraform.io/hashicorp/kubernetes]
├── module.gcp-network
│   ├── provider[registry.terraform.io/hashicorp/google] >= 2.12.0, < 4.0.0
│   ├── module.firewall_rules
│   │   └── provider[registry.terraform.io/hashicorp/google] >= 2.12.0, < 4.0.0
│   ├── module.routes
│   │   └── provider[registry.terraform.io/hashicorp/google] >= 2.12.0, < 4.0.0
│   ├── module.subnets
│   │   └── provider[registry.terraform.io/hashicorp/google] >= 2.12.0, < 4.0.0
│   └── module.vpc
│       ├── provider[registry.terraform.io/hashicorp/google] ~> 3.45
│       └── provider[registry.terraform.io/hashicorp/google-beta] ~> 3.45
└── module.gke
    ├── provider[registry.terraform.io/hashicorp/google-beta] >= 3.79.0, < 4.0.0
    ├── provider[registry.terraform.io/hashicorp/kubernetes] ~> 2.0
    ├── provider[registry.terraform.io/hashicorp/google]
    ├── provider[registry.terraform.io/hashicorp/random]
    └── module.gcloud_delete_default_kube_dns_configmap
        └── module.gcloud_kubectl
            ├── provider[registry.terraform.io/hashicorp/external]
            ├── provider[registry.terraform.io/hashicorp/null]
            └── provider[registry.terraform.io/hashicorp/random]

Providers required by state:

    provider[registry.terraform.io/hashicorp/google]

    provider[registry.terraform.io/hashicorp/random]
```

### Terraform Version

```sh
Terraform v1.0.11
on darwin_amd64
+ provider registry.terraform.io/hashicorp/external v2.1.0
+ provider registry.terraform.io/hashicorp/google v3.90.0
+ provider registry.terraform.io/hashicorp/google-beta v3.90.0
+ provider registry.terraform.io/hashicorp/kubernetes v2.6.1
+ provider registry.terraform.io/hashicorp/null v3.1.0
+ provider registry.terraform.io/hashicorp/random v3.1.0
```

**Comment by morgante** [comment]
Check your Terraform provider config. It seems likely you have the `google` and `google-beta` providers configured differently (please paste your config).

**Comment by jpresky-seegrid** [comment]
provider.tf:
terraform {
  required_providers {
    google = {
      source  = "hashicorp/google"
    }
  }
  backend "gcs" {
    credentials = "file.json"
    bucket      = "company-tfstate"
    prefix      = "terraform-gke"
  }
}
provider "google" {
  credentials = file("file.json")
}

main.tf
data "google_client_config" "default" {
}

provider "kubernetes" {
  host                   = "https://${module.gke.endpoint}"
  cluster_ca_certificate = base64decode(module.gke.ca_certificate)
}

module "gcp-network" {
  source       = "terraform-google-modules/network/google"
  version      = "~> 3.1"
  project_id   = var.project_id
  network_name = var.network

  subnets = [
    {
      subnet_name           = var.subnetwork
      subnet_ip             = "10.1.0.0/18"
      subnet_region         = var.region
      subnet_private_access = "true"
    },
  ]

  secondary_ranges = {
    (var.subnetwork) = [
      {
        range_name    = var.ip_range_pods_name
        ip_cidr_range = "10.2.0.0/16"
      },
      {
        range_name    = var.ip_range_services_name
        ip_cidr_range = "10.1.64.0/18"
      },
    ]
  }
}

data "google_compute_subnetwork" "subnetwork" {
  name       = var.subnetwork
  project    = var.project_id
  region     = var.region
  depends_on = [module.gcp-network]
}

module "gke" {
  source     = "terraform-google-modules/kubernetes-engine/google//modules/beta-private-cluster-update-variant"
  project_id = var.project_id
  name       = var.cluster_name
  regional   = true
  region     = var.region
  zones      = slice(var.zones, 0, 1)


  network                 = module.gcp-network.network_name
  subnetwork              = module.gcp-network.subnets_names[0]
  ip_range_pods           = var.ip_range_pods_name
  ip_range_services       = var.ip_range_services_name
  create_service_account  = true
  enable_private_endpoint = true
  enable_private_nodes    = true
  master_ipv4_cidr_block  = "172.16.0.0/28"
  #datapath_provider       = "ADVANCED_DATAPATH"

  master_authorized_networks = [
    {
      cidr_block   = data.google_compute_subnetwork.subnetwork.ip_cidr_range
      display_name = "VPC"
    },
  ]

  http_load_balancing        = true
  horizontal_pod_autoscaling = true
  network_policy             = true


  database_encryption = [{
    state    = "ENCRYPTED"
    key_name = google_kms_crypto_key.cloudops-cluster-key.id
  }]

  node_pools = [
      {
        name                      = "default-pool"
        machine_type              = "n1-standard-8"
        node_locations            = "us-east4-a"
        min_count                 = 0
        max_count                 = 3
        local_ssd_count           = 1
        disk_size_gb              = 100
        disk_type                 = "pd-standard"
        image_type                = "COS"
        auto_repair               = true
        auto_upgrade              = false
        service_account           = "default"
        preemptible               = false
        initial_node_count        = 3
      }
      ]
}

**Comment by morgante** [comment]
There's your problem. You haven't configured the beta provider as well: https://registry.terraform.io/providers/hashicorp/google/latest/docs/guides/provider_versions#using-the-google-beta-provider

just add this to your providers config:
```
provider "google" {
   credentials = file("file.json")
}

provider "google-beta" {
   credentials = file("file.json")
}
```

**Comment by jpresky-seegrid** [comment]
ahh thanks!!!
